nasa-gibs/worldview
# WV-2493 Charting Mode

## Description
Create interactive prototype in Worldview of rapid analysis imagery statistics UI that demonstrate end-to-end capabilities of imagery statistics. Initial work has been done utilizing the PO.DAAC ImageStat API. We would like to include this preliminary implementation into the release behind a feature flag (disabled by default).

Frontend charting implementation should be designed so that different charting APIs, e.g PO.DAAC imagestat, VEDA, SentinelHub Statistical API, can be used to display the same types of information without needing to have a substantially different core charting codebase.

NOTE: This feature is disabled by default. To enable, open `/config/default/common/features.json` & set `charting `to `true` & then run `npm run build`.

## How To Test
NOTE: This API often has CORS errors that prevent success from localhost. I have been using the CORS Everywhere extension for Firefox to get around that problem. 'Allow CORS' extension for Chrome also works. You should be on the VPN as well.

- git checkout WV-2493-simple-statistics
- npm install
- npm run build (if you change the charting mode availability)
- npm run watch
- Add any layers with a palette (for instance, Aqua/Airs Carbon Monoxide (L2, 500 hPa, Night) or Aura/OMI Aerosol Index). 
- Click the Start Charting button to test the feature.

## Usage:
- Click the pencil to draw an Area of Interest (AOI). If you do not specify an AOI, it will default to the full map.
- Select a date. On start of chart mode, the date from the timeline will be your start date. Click the calendar icon to bring up the charting mode date selector.

NOTE: You should keep your Networking tab open as the API sometimes (regularly) responds with an internal error 500. In that case, you can try changing your AOI and/or reducing your time range to increase your odds of success.

- 'Single Date' mode --> Modal with mean/median/max/stddev for that date. Takes 3-5 seconds typically.
- 'Date Range ' mode--> Dynamic chart of mean/median/max/stddev across 10 steps from the start date to end date. This can easily take 30 seconds + to get data.

@nasa-gibs/worldview
